packages/platforms/electron-player/src/main/api/file-system.ts

```diff
@@ -0,0 +1,214 @@
+  storagePath: string,
+  url: string,
+  data = null
+): Promise<StoreFileReturnValue> {
+  // Must be imported dynamically since the main process is a CommonJS module but @castmill/cache is an ES module
+  const { StoreResult } = await import('@castmill/cache');
+
+  const fullPath = join(BASE_DIR, storagePath);
+  try {
+    const filename = getFileName(url);
+    const filePath = join(fullPath, filename);
+    if (data) {
+      await writeFile(filePath, data);
+    } else {
+      await downloadFile(url, filePath);
+    }
+    const stats = await stat(filePath);
+    return {
+      result: { code: StoreResult.Success },
+      item: {
+        url: filePath,
+        size: stats.size,
+      },
+    };
+  } catch (error: any) {
+    console.error('Failed to store file:', error);
+    const errMsg = error?.message ?? 'Unknown Error';
+    return {
+      result: { code: StoreResult.Failure, errMsg },
+    };
+  }
+}
+
+/*
+ * Retrieve a file from the storage. Returns the file path if the file exists,
+ */
+export async function retrieveFile(
+  storagePath: string,
+  url: string
+): Promise<string | void> {
+  const fullPath = join(BASE_DIR, storagePath);
+  try {
+    const filePath = join(fullPath, getFileName(url));
+    await stat(filePath); // Check if file exists
+    return filePath;
+  } catch (error) {
+    console.log('Failed to retrieve file:', error);
+    return undefined; // File does not exist
+  }
+}
+
+/*
+ * Delete a file from the storage
+ */
+export async function deleteFile(
+  storagePath: string,
+  url: string
+): Promise<void> {
+  const fullPath = join(BASE_DIR, storagePath);
+  try {
+    const filePath = join(fullPath, getFileName(url));
+    await unlink(filePath);
+  } catch (error) {
+    console.error('Failed to delete file:', error);
+    throw error;
+  }
+}
+
+/*
+ * Delete all files from the storage
+ */
+export async function deleteAllFiles(storagePath: string): Promise<void> {
+  const fullPath = join(BASE_DIR, storagePath);
+  try {
+    const files = await readdir(fullPath);
+    await Promise.all(files.map((file) => unlink(join(fullPath, file))));
+  } catch (error) {
+    console.error('Failed to delete all files:', error);
+  }
+}
+
+/*
+ * Download a file from the URL and save it to the destination path
+ */
+function downloadFile(url: string, destPath: string): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const request = net.request(url);
+    request.on('response', (response) => {
+      if (response.statusCode !== 200) {
+        console.error(
+          `Failed to download file, status code: ${response.statusCode}`
+        );
+        return reject(
+          Error(`Failed to download file, status code: ${response.statusCode}`)
+        );
+      }
+      const writeStream = createWriteStream(destPath);
+      response.on('data', (chunk) => {
+        writeStream.write(chunk);
+      });
+      response.on('end', () => {
+        writeStream.end();
```

Comment:
a handler for error is missing for both request and response.

Reply:
Will fix

Reply:
fixed